### PR TITLE
What packages to retire entirely

### PR DESCRIPTION
**Pull request by hroncok**
@encukou Could you please review the logic of this? The code is far from ideal (I've copy pasted entire blocks), but for a one time automation, I don't think I care about code quality (that much at least). The thing that needs review is: **Are the packages marked for entire retirement OK to be retired?**

Thanks.

### CODE REVIEW

**Comment by hroncok** [comment]
The results seemed quite surprising:

Binary packages:
1100: drop_later
911: drop_now
751: keep
166: investigate

Source packages:
1096: retire_later
1059: keep
773: retire_now

-- the numbers are not representing source packages, but the number of binary packages where we will retire the whole srpm. I'll change the numbers.

**Comment by hroncok** [comment]
Added a fixup.

Binary packages:
1100: drop_later
911: drop_now
751: keep
166: investigate

Source packages:
1020: retire_later
843: keep
758: retire_now

File: portingdb/check_drops.py [hunk 1]
```diff
@@ -461,3 +533,18 @@ def check_drops(ctx, filelist, cache_sax, cache_rpms):
...
+    log('\nSource packages:')
+    stats_counter = collections.Counter(r['source_verdict'] for r in results.values())
+    for package, number in stats_counter.most_common():
+        log('{}: {}'.format(number, package))
+
+    for verdict in stats_counter:
+        if verdict == 'keep':
+            continue
+        filtered = {n: r for n, r in results.items() if r['source_verdict'] == verdict}
+        with open(cache_dir / ('results-' + verdict + '.json'), 'w') as f:
```
Comment (encukou) [inline portingdb/check_drops.py]: Add `-srpms` here as well? I don't think we want to rewrite the earlier files.

Comment (hroncok) [reply portingdb/check_drops.py]: The verdict names differ and the data are collected by binary packages.

Comment (hroncok) [reply portingdb/check_drops.py]: the verdicts names I mean

**Comment by encukou** [comment]
> Source packages:
> 1020: retire_later
> 843: keep
> 758: retire_now

Hmm. This doesn't seem right.
Portingdb lists 63 Mispackaged + 856 Idle +  311 Blocked + 90 Dropped packages. That's less than 1320 py2-only packages.
But retire_later+retire_now adds up to 1778. Where are all those py2 packages coming from?

File: portingdb/check_drops.py (hunk 1 above)
Comment (encukou) [reply portingdb/check_drops.py]: Ah. Still, consider adding it just to make things clearer.

**Comment by hroncok** [comment]
<del>Hmm, maybe I should check if we even have some packages before I check if all have some thing.</del>

So apparently this marked `file` for removal, I'll try to figure out why.

**Comment by hroncok** [comment]
Source packages:
2355: keep
136: retire_now
130: retire_later

Sounds a bit less dramatic.

File: portingdb/check_drops.py (hunk 1 above)
Comment (hroncok) [reply portingdb/check_drops.py]: ok

**Comment by encukou** [comment]
Looks good now, thanks!

**Pull request merged**